### DISCUSSION TITLE
CUSTOM METHOD ->The POST method is not supported for this route. Supported methods: GET, HEAD.

### DISCUSSION THREAD

**Discussion by edmaSB** [orchidsoftware/platform#2772] [Q&A]
HI,
I have a screen with 2 methods
`  
public function partecipa(Torneo $torneo, Request $request)
    {
        $torneo
            ->torneiMy()
            ->create(['users_id' => $request->user()->id, 'tornei_id' => $torneo->id]);

        redirect()->route('platform.torneo.torneiMy');
    }
    public function esci(Torneo $torneo, Request $request)
    {

        if (strtotime($torneo->fine_iscrizioni) > time()) {

            $myTorneo = Torneo::where('users_id', $request->user()->id)->where('tornei_id', $torneo->id)->firstOrFail();

            $torneo
                ->torneiMy()
                ->dissociate($myTorneo);
            Toast::success(__('Sei uscito dal torneo'));
        } else {

            Toast::error(__('Il tempo di iscrizione é terminato non puoi uscire dal torneo'));
        }

        redirect()->route('platform.torneo.torneiMy');
    }`

 for the "esci" method gives me an error.

**The POST method is not supported for this route. Supported methods: GET, HEAD.**

How can I solve it?

**Comment by c-gross** [comment]
How does your route look like to call that screen? (routes/platform.php)

**Reply by edmaSB** [reply]
Route::screen('torneiMy', TorneoMyScreen::class)
->name('platform.torneo.torneiMy')
->breadcrumbs(fn (Trail $trail) => $trail
    ->parent('platform.index')
    ->push(__('I miei Tornei'), route('platform.torneo.torneiMy')));



Button::make(__('Partecipa al Torneo'))
                                        ->icon('bs.check-circle')
                                        //->confirm(__('ATTENZIONE! Eliminare un torneo elimina anche tutti i dati associati compresi i risultati dei giocatori. Continuare?'))
                                        ->method('esci', [
                                            'torneo' => $torneoMy->id,
                                        ]),

**Reply by c-gross** [reply]
Ah it's a button method call inside the screen. That does not require a specific route entry. 
Just tested, it's a POST request and is working for me.

**Reply by edmaSB** [reply]
not for me. only one method work for me for single screen

**Comment by edmaSB** [comment]
Can anyone help me?

**Reply by tartesso26** [reply]
did you find a solution for this?? i have same problem

**Comment by tabuna** [comment]
Since a single screen declaration is used in routes (for both GET and POST requests), the order of their declaration matters. You can find an example here:
https://orchid.software/en/docs/screens/#routing-screens
